core/runtime/jsrealm.rs

```diff
@@ -75,10 +75,16 @@ pub(crate) struct ContextState<OpDriverImpl: OpDriver = DefaultOpDriver> {
 
 impl<O: OpDriver> ContextState<O> {
   pub(crate) fn new(
+    op_driver: Rc<O>,
     isolate_ptr: *mut v8::OwnedIsolate,
     get_error_class_fn: GetErrorClassFn,
-    ops_with_metrics: Vec<bool>,
+    op_ctxs: Box<[OpCtx]>,
```

Comment:
Look - no more entanglement between `ContextState` and `OpCtx` - now we can pass fully initialized ops here and remove a lot of complexity and spaghetti code.